src/main/java/org/isf/mortuarystays/manager/MortuaryStayManager.java

```diff
@@ -0,0 +1,154 @@
+	 */
+	public List<MortuaryStay> getAll() throws OHServiceException {
+		return ioOperations.getAll();
+	}
+
+	/**
+	 * Updates the specified {@link MortuaryStay}.
+	 * @param mortuary - the {@link MortuaryStay} to update.
+	 * @return mortuary that has been updated.
+	 * @throws OHServiceException if an error occurs during the update.
+	 */
+	public MortuaryStay update(MortuaryStay mortuary) throws OHServiceException {
+		return ioOperations.update(mortuary);
+	}
+
+	/**
+	 * Stores the specified {@link MortuaryStay}.
+	 * @param mortuary the mortuary to store.
+	 * @return mortuary that has been stored.
+	 * @throws OHServiceException if an error occurs storing the mortuary.
+	 */
+	public MortuaryStay add(MortuaryStay mortuary) throws OHServiceException {
+		validate(mortuary, true);
+		return ioOperations.add(mortuary);
+	}
+
+	/**
+	 * Deletes a {@link MortuaryStay} in the DB.
+	 * @param mortuary - the item to delete
+	 * @throws OHServiceException
+	 */
+	public void delete(MortuaryStay mortuary) throws OHServiceException {
+		ioOperations.delete(mortuary);
+	}
+
+	/**
+	 * Checks if the code is already in use.
+	 * @param code - the {@link MortuaryStay} code
+	 * @return {@code true} if the code is already in use, {@code false} otherwise
+	 * @throws OHServiceException
+	 */
+	public boolean isCodePresent(String code) throws OHServiceException {
+		return ioOperations.isCodePresent(code);
+	}
+
+	/**
+	 * Returns the {@link MortuaryStay} based on vaccine type code.
+	 * @param code - the  {@link MortuaryStay} code.
+	 * @return the {@link MortuaryStay} or {@literal null} if none found
+	 * @throws OHServiceException
+	 */
+	public MortuaryStay getByCode(String code) throws OHServiceException {
+		if (code == null) {
+			throw new OHDataValidationException(new OHExceptionMessage(MessageBundle.getMessage("angal.common.pleaseinsertacode.msg")));
+		}
+		return ioOperations.getByCode(code);
+	}
+
+	/**
+	 * Verify if the object is valid for CRUD and return a list of errors, if any.
+	 * @param mortuaryStay the {@link MortuaryStay} object to validate.
+	 * @param insert {@code true} or updated {@code false}
+	 * @throws OHServiceException
+	 */
+	protected void validate(MortuaryStay mortuaryStay, boolean insert) throws OHServiceException {
+		String code = mortuaryStay.getCode();
+		String desc = mortuaryStay.getDescription();
+		String name = mortuaryStay.getName();
+		int minD = mortuaryStay.getMinDays();
+		int maxD = mortuaryStay.getMaxDays();
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		if (code.isEmpty()) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.common.pleaseinsertacode.msg")));
+		}
+		if (code.length() > 11) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.common.thecodeistoolongmax1char.msg")));
+		}
+
+		if (name.isEmpty()) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.mortuarystays.pleaseinsertavalidname.msg")));
+		}
```

Comment:
Avoid this kind of validation. If you really wand to validate thoes field there, then you could annotate your entity fields with Validation annotations and use an validatio factoctotry to validate the entire class.
Eg:
`
@Column(name = "MRTST_CODE")
@Size(min = 11, message = "angal.mortuarystays.*.msg")
private String code;

@Column(name = "MRTST_MIN_DAYS")
@Positive(message = "angal.mortuarystays.*.msg")
private String minDays;

// Later in your code

Set<ConstraintViolation<MortuaryStay>> violations = validator.validate(mortuaryStay);
//throw some error with the constraint violations, while translating them.
`

Reply:
I think we can live it like that. to use the same way like officiel version